src/core/email-client.js

```diff
@@ -53,102 +72,241 @@ function getFreshImap (cb) {
-                var collectUnseenEmails = function () {
-                    self.imap.search(['UNSEEN'], function (err, results) {
-                        if (err || results.length === 0) {
-                            console.log(err || 'NO NEW MAIL');
-                            return;
-                        }
-
-                        //  Fetch the entire email bodies
-                        var f = self.imap.fetch(results, {
-                            bodies: [''],
-                            markSeen: true
-                        });
+    //attachments as array of Path
+    _sendEmail(toName, toEmail, subject, text, attachments){
+        var self = this;
+        var fromText = {
+            name: self._name,
+            address: self._email
+        };
+        var toText = {
+            name: toName, 
+            address: toEmail
+        };
+
+        var files = [];
+        if (attachments !== null){
+            for (var i = 0; i < attachments.length; i++){
+                
+                files.push({
+                    path: attachments[i]
+                });
+            }
+        }
 
-                        f.on('message', function (msg, seqno) {
-                            var parser = new MailParser();
-                            parser.on('end', function (mail) {
-                                //  Done parsing
-                                //  TODO: logic for sending event to event bus goes here
-                                console.log('NEW MAIL!');
-                                console.log(
-                                    'From: ' + mail.from[0].address + '\n' +
-                                    'Subject: ' + mail.subject + '\n' +
-                                    'Body:\n' + mail.text
-                                );
+        //mail component
+        var mailOptions = {
+            from: fromText, // sender address
+            to: toText, // list of receivers
+            subject: subject, // Subject line
+            text: text, // plaintext body
+            attachments: files //attachments
+        }
+
+        // send mail with defined transport object
+        self._loadTransporter().done(function (transporter){
+            self._transporter = transporter;
+            // send mail with defined transport object
+            self._transporter.sendMail(mailOptions, function(error, info){
+                if(error){
+                    console.log(error);
+                    return;
+                }
+                self._SendCheck.resolve(true);
+                console.log('Message sent: ' + info.response);
+                var content = {
+                    fromEmail: fromText.address,
+                    fromName: fromText.name,
+                    subject: subject,
+                    body: text
+                };
+                self._eventBus.emitEvent(eventObj['SEND_EMAIL'], content);
```

Comment:
SEND_EMAIL is for other components to tell the email client to send an email. It shouldn't be emitting this event to itself